src/Workleap.Extensions.Mongo/Indexing/IndexRegistry.cs

```diff
@@ -6,8 +6,14 @@ namespace Workleap.Extensions.Mongo.Indexing;
 /// <summary>
 /// Associates a concrete <see cref="IMongoDocument"/> class with its single <see cref="MongoIndexProvider"/>.
 /// </summary>
-internal sealed class IndexRegistry : Dictionary<Type, Type>
+internal sealed class IndexRegistry : Dictionary<string, DocumentTypeEntry>
```

Comment:
When designing the library two years ago, I made a mistake. I believed that developers would only declare ONE single C# class for exactly ONE MongoDB collection.

Time has shown that developers can create as many C# document types that could be mapped to a same collection, as long as they have a similar shape. Most of the time, they only contain a subset of the properties.

I never thought that someone would use inheritance, combined with the index provider, to target a same collection.

So this `IndexRegistry` was fundamentally wrong by assuming that its dictionary key was the C# document type. It's supposed to be the collection name. Only the collection is unique. Also here now, we make sure that for one collection, there's only one index provider type as well.